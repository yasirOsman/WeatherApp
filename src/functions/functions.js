//gets the maximum temperature from the provided data of given date
const getMaxTemp = (data,date) => {
    let max_Temp = Number.MIN_SAFE_INTEGER;
    for(let i = 0; i < data.length; i++){
        let current = data[i];
        let currentDate = current.dt_txt.split(" ");
        if(currentDate[0] === date ){
            if(current.main.temp_max > max_Temp){
            max_Temp = current.main.temp_max;
            }
        }
    }
    return (max_Temp -273.15).toFixed(2);
}

//gets the next 5 unique dates
const getDate = (data) => {
   let dates = [];
       for (let i = 0; i < data.length - 1; i++) {
           let currentDate = data[i].dt_txt.split(" ");
           let nextDate = data[i+1].dt_txt.split(" ");
           if (currentDate[0] != nextDate[0] && !dates.find(element => element === currentDate[0]) && dates.length != 5) {
               dates.push(currentDate[0]);
            }
            if(!dates.find(element => element === nextDate[0]) && dates.length != 5){
                dates.push(nextDate[0]);
            }
       }
   return dates;
}

//gets the average humidity of the given date
const getAverageHumidity =(data,date) => {
    let average = 0;
    let cnt = 0;
    for(let i = 0; i < data.length; i++){
        let current = data[i];
        let currentDate = current.dt_txt.split(" ");
        if(currentDate[0] === date ){
            average += current.main.humidity;
            cnt++;
        }
    }
    average = average / cnt;
    return average.toFixed(2);
}

export {getMaxTemp, getDate, getAverageHumidity}
