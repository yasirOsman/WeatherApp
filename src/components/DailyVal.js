import React from 'react';
import {getMaxTemp, getAverageHumidity} from '../functions/functions';

const Dailyval = (props) => {
    console.log(props.theList)
    const values = props.theList.map(element => 
            
            <tr key={element}>         
                <td>{element}</td>
                <td>{getMaxTemp(props.list,element)}Â°</td>
                <td>{getAverageHumidity(props.list,element)}%</td>
             </tr>
            
        
        )
        // console.log(props.list)
        return values
    
}

export default Dailyval;