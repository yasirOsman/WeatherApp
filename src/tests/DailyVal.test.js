import React from 'react';
import {shallow} from 'enzyme'
import DailyVal from '../components/DailyVal';
import {data} from './testData';
import {getDate} from '../functions/functions';

test('Should render DailyVal with rows filled with the date, max temp and avg humidity', () => {
    const wrapper = shallow(<DailyVal theList={getDate(data.list)} list={data.list}/>);
    expect(wrapper).toMatchSnapshot();
})

