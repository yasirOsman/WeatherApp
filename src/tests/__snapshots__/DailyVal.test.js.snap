// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DailyVal with rows filled with the date, max temp and avg humidity 1`] = `
Array [
  <tr
    key="11/12/2018"
/>,
  <tr
    key="13/12/2018"
/>,
  <tr
    key="14/12/2018"
/>,
]
`;
